src/veranda/raster/mosaic/netcdf.py

```diff
@@ -0,0 +1,595 @@
+            True if the NetCDF file(s) should be overwritten, False if not (default).
+        unlimited_dims : list of str or str, optional
+            List of dimension names specifying the dimensions to be stored as unlimited.
+        kwargs : dict, optional
+            Key-word arguments for creating a `NetCdf4File` instance.
+
+        """
+        unlimited_dims = to_list(unlimited_dims)
+        data_variables = to_list(data_variables) if data_variables is not None else list(self._data.data_vars)
+        data = self.data[data_variables]
+
+        all_dims = list(data.dims)
+        space_dims = all_dims[-2:]
+        stack_dim_names = all_dims[:-2]
+
+        nodatavals = dict()
+        scale_factors = dict()
+        offsets = dict()
+        dtypes = dict()
+        for data_variable in data_variables:
+            dtypes[data_variable] = data[data_variable].data.dtype.name
+            nodatavals[data_variable] = data[data_variable].attrs.get('_FillValue', 0)
+            scale_factors[data_variable] = data[data_variable].attrs.get('scale_factor', 1)
+            offsets[data_variable] = data[data_variable].attrs.get('add_offset', 0)
+
+        for filepath, file_group in self._file_register.groupby('filepath'):
+            tile_id = file_group.iloc[0].get('tile_id', '0')
+            if apply_tiling:
+                tile = self._mosaic[tile_id]
+                if not tile.intersects(self._data_geom):
+                    continue
+                tile_write = self._data_geom.slice_by_geom(tile, inplace=False)
+                data_write = data.sel(**{space_dims[0]: tile_write.y_coords, space_dims[1]: tile_write.x_coords})
+            else:
+                tile_write = self._data_geom
+                tile = self._data_geom
+                data_write = data
+
+            layer_ids = np.unique(file_group[self._file_dim])
+            data_write = data_write.sel(**{self._file_dim: layer_ids})
+
+            stack_dims = dict()
+            for stack_dim_name in stack_dim_names:
+                if stack_dim_name in unlimited_dims:
+                    stack_dims[stack_dim_name] = None
+                else:
+                    stack_dims[stack_dim_name] = len(data_write[stack_dim_name])
+
+            with NetCdf4File(filepath, mode='w', data_variables=data_variables, stack_dims=stack_dims,
+                             space_dims={space_dims[0]: tile.n_rows, space_dims[1]: tile.n_cols},
+                             scale_factors=scale_factors, offsets=offsets, nodatavals=nodatavals, dtypes=dtypes,
+                             attrs={self._file_dim: {'units': 'days since 1950-01-01 00:00:00'}},  # TODO: make this more flexible (this needs to be defined from outside)
+                             geotrans=tile_write.geotrans,
+                             sref_wkt=tile_write.sref.wkt, metadata=data_write.attrs, **kwargs) as nc_file:
+                nc_file.write(data_write, encoder=encoder, encoder_kwargs=encoder_kwargs)
```

Comment:
Again almost the same as `write` and also almost the same as `GeoTiff::write`. You could collapse all those functions into common parameterizeable chunks and massively reduce code duplication

Reply:
Done.